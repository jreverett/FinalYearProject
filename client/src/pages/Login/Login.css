#form-container {
  position: fixed;
  top: 50%;
  transform: translate(0%, -50%);
  -webkit-transform: translate(0%, -50%);
  -moz-transform: translate(0%, -50%);
  -o-transform: translate(0%, -50%);
  -ms-transform: translate(0%, -50%);

  padding: 20px;
  background-color: rgb(248, 248, 248);
  border-radius: 10px;
}

.help-block {
  font-size: small;
  color: rgb(255, 82, 62);
}

.compulsory-asterisk {
  display: inline;
  color: rgb(255, 82, 62);
}

#logo-container {
  width: 50%;
  margin: 0 auto;
}

#logo {
  padding-bottom: 20px;
}

.form-group {
  margin-bottom: 10px;
}

#signup-button-container {
  width: 50%;
  margin: 0 auto;
}

#signup-button {
  margin-top: 20px;
  width: 100%;
}

.strike {
  display: block;
  text-align: center;
  overflow: hidden;
  white-space: nowrap;
}

.strike > span {
  position: relative;
  display: inline-block;
}

.strike > span:before,
.strike > span:after {
  content: '';
  position: absolute;
  top: 50%;
  width: 9999px;
  height: 1px;
  background: black;
}

.strike > span:before {
  right: 100%;
  margin-right: 15px;
}

.strike > span:after {
  left: 100%;
  margin-left: 15px;
}

#question-container {
  text-align: center;
  padding-top: 20px;
  padding-bottom: 10px;
}

#question-container p {
  display: inline;
}

#question-container a {
  color: blue;
  font-size: unset;
}

#loading-wheel {
  position: absolute;
  top: 50%;
  left: calc(50% - 25px);
  width: 50px;
}

#loading-fade {
  position: absolute;
  width: 100vw;
  height: 100vh;
  background-color: rgba(230, 230, 230, 0.5);
  z-index: 10;
}
