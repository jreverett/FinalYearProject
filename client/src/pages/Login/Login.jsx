import React, { Component, Fragment } from 'react';
import { authenticationService } from '../../services';
import '../../common.css';
import './Login.css';

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      submitted: false,
      loading: false,
      error: ''
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  handleSubmit(e) {
    e.preventDefault();

    this.setState({ submitted: true });
    const { email, password } = this.state;

    // check form is valid
    if (!(email && password)) return;

    this.setState({ loading: true });
    authenticationService.login(email, password).then(
      user => {
        this.setState({ loading: false });
      },
      error => this.setState({ error, loading: false })
    );
  }

  componentWillUpdate() {
    // if user is already logged in, redirect to home
    if (this.props.loggedInUser) window.location.href = '/';
  }

  render() {
    const { submitted, email, password, loading, error } = this.state;
    return (
      <Fragment>
        {loading && <div id="loading-fade" />}

        <div id="form-container" className="col-md-4 offset-md-4">
          <div id="logo-container">
            <img
              id="logo"
              src={require('../../images/upvent.png')}
              alt="Upvent logo"
            />
          </div>

          <form name="form" onSubmit={this.handleSubmit}>
            {/* EMAIL */}
            <div
              className={
                'form-group' + (submitted && !email ? ' has-error' : '')
              }
            >
              <label htmlFor="email">
                Email<p className="compulsory-asterisk">*</p>
              </label>
              <input
                type="text"
                className="form-control"
                name="email"
                value={email}
                onChange={this.handleChange}
              />
              {submitted && !email && (
                <div className="help-block">Email is required</div>
              )}
            </div>

            {/* PASSWORD */}
            <div
              className={
                'form-group' + (submitted && !password ? ' has-error' : '')
              }
            >
              <label htmlFor="password">
                Password<p className="compulsory-asterisk">*</p>
              </label>
              <input
                type="password"
                className="form-control"
                name="password"
                value={password}
                onChange={this.handleChange}
              />
              {submitted && !password && (
                <div className="help-block">Password is required</div>
              )}
            </div>
            <div id="login-button-container" className="form-group">
              <button
                id="login-button"
                className="btn btn-primary"
                disabled={loading}
              >
                Login
              </button>
              {loading && (
                <Fragment>
                  <img
                    id="loading-wheel"
                    src="data:image/gif;base64,R0lGODlhEAAQAPIAAP///wAAAMLCwkJCQgAAAGJiYoKCgpKSkiH/C05FVFNDQVBFMi4wAwEAAAAh/hpDcmVhdGVkIHdpdGggYWpheGxvYWQuaW5mbwAh+QQJCgAAACwAAAAAEAAQAAADMwi63P4wyklrE2MIOggZnAdOmGYJRbExwroUmcG2LmDEwnHQLVsYOd2mBzkYDAdKa+dIAAAh+QQJCgAAACwAAAAAEAAQAAADNAi63P5OjCEgG4QMu7DmikRxQlFUYDEZIGBMRVsaqHwctXXf7WEYB4Ag1xjihkMZsiUkKhIAIfkECQoAAAAsAAAAABAAEAAAAzYIujIjK8pByJDMlFYvBoVjHA70GU7xSUJhmKtwHPAKzLO9HMaoKwJZ7Rf8AYPDDzKpZBqfvwQAIfkECQoAAAAsAAAAABAAEAAAAzMIumIlK8oyhpHsnFZfhYumCYUhDAQxRIdhHBGqRoKw0R8DYlJd8z0fMDgsGo/IpHI5TAAAIfkECQoAAAAsAAAAABAAEAAAAzIIunInK0rnZBTwGPNMgQwmdsNgXGJUlIWEuR5oWUIpz8pAEAMe6TwfwyYsGo/IpFKSAAAh+QQJCgAAACwAAAAAEAAQAAADMwi6IMKQORfjdOe82p4wGccc4CEuQradylesojEMBgsUc2G7sDX3lQGBMLAJibufbSlKAAAh+QQJCgAAACwAAAAAEAAQAAADMgi63P7wCRHZnFVdmgHu2nFwlWCI3WGc3TSWhUFGxTAUkGCbtgENBMJAEJsxgMLWzpEAACH5BAkKAAAALAAAAAAQABAAAAMyCLrc/jDKSatlQtScKdceCAjDII7HcQ4EMTCpyrCuUBjCYRgHVtqlAiB1YhiCnlsRkAAAOwAAAAAAAAAAAA=="
                  />
                </Fragment>
              )}
            </div>
            {error && <div className={'alert alert-danger'}>{error}</div>}
          </form>
          <br />
          <div className="strike">
            <span>or</span>
          </div>
          <div id="question-container">
            <p>Don't have an account? </p>
            <a href="/signup">Sign up</a>
          </div>
        </div>
      </Fragment>
    );
  }
}

export default Login;
