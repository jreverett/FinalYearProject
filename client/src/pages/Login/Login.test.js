import React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';

describe('<Login />', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
