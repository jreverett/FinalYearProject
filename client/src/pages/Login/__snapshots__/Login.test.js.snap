// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches the snapshot 1`] = `
<div
  className="form-container col-md-4 offset-md-4"
>
  <div
    id="logo-container"
  >
    <img
      alt="Upvent logo"
      id="logo"
      src="upvent.png"
    />
  </div>
  <form
    name="form"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="email"
      >
        Email
        <p
          className="compulsory-asterisk"
        >
          *
        </p>
      </label>
      <input
        autoComplete="username"
        className="form-control"
        name="email"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        htmlFor="password"
      >
        Password
        <p
          className="compulsory-asterisk"
        >
          *
        </p>
      </label>
      <input
        autoComplete="current-password"
        className="form-control"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
      id="login-button-container"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        id="login-button"
      >
        Login
      </button>
    </div>
    
  </form>
  <div
    className="login-forgot-password-container"
  >
    <a
      className="login-forgot-password"
      href="/forgot-password"
    >
      Forgot password?
    </a>
  </div>
  <div
    className="strike"
  >
    <span>
      or
    </span>
  </div>
  <div
    id="question-container"
  >
    <p>
      Don't have an account? 
    </p>
    <a
      href="/signup"
    >
      Sign up
    </a>
  </div>
</div>
`;
